src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -253,6 +253,11 @@ default ProgrammingExercise findOneByProjectKeyOrThrow(String projectKey, boolea
             """)
     Optional<ProgrammingExercise> findByParticipationId(@Param("participationId") Long participationId);
 
+    default ProgrammingExercise findByParticipationIdOrElseThrow(long participationId) throws EntityNotFoundException {
+        return findByParticipationId(participationId)
+                .orElseThrow(() -> new EntityNotFoundException("Programming exercise for participation with id " + participationId + " does not exist"));
```

Comment:
I do not really agree because if I use one of the other ones, The message will be something such as ProgrammingExercise with id 123 not found which doesn't really reflect the query.